### ISSUE TITLE
Surface Laptop Go 1 broken suspend resume on mainline kernel>6.5

### ISSUE THREAD

**Issue by snizzo** [linux-surface/linux-surface#1306]
### Description of the bug or feature

Whenever I close the lid, laptop goes in standby mode (s0ix?). When I open it, it's like frozen for 30 secs / 1 minute before appearing again, sometimes broken with no touchpad, sometimes completely frozen.

This behavious is not present with the same software running (de, distribution etc...) but with a different kernel from the previous version. Kernel is the standard mainline shipped with ubuntu. It used to work perfectly until now. So:

6.5.0-13-generic -> not working
6.2.0-36-generic -> working

I compared the 2 dmesg from the 2 different kernel. The 6.2 one says it's all working fine. 6.5 shows quite some error messages I'm having a bit of a hard time with.

### Environment

Ubuntu 23.10 (mantic)
OpenGL renderer string: Mesa Intel(R) UHD Graphics (ICL GT1)

- Hardware model: Surface Laptop Go 1
- Kernel version: Linux my-Surface-Laptop-Go 6.5.0-13-generic # 13-Ubuntu SMP PREEMPT_DYNAMIC Fri Nov  3 12:16:05 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
- Distribution: Ubuntu 23.10 (mantic)

<details><summary>`dmesg` output</summary>

```
[dmesg_brokensuspend.txt](https://github.com/linux-surface/linux-surface/files/13445144/dmesg_brokensuspend.txt)
[   12.872629] wlp0s20f3: disconnect from AP c2:a5:11:2e:49:3f for new auth to bc:a5:11:2e:49:42
[   12.926330] wlp0s20f3: authenticate with bc:a5:11:2e:49:42
[   12.933221] wlp0s20f3: send auth to bc:a5:11:2e:49:42 (try 1/3)
[   12.961122] wlp0s20f3: authenticated
[   12.963489] wlp0s20f3: associate with bc:a5:11:2e:49:42 (try 1/3)
[   13.063620] wlp0s20f3: RX ReassocResp from bc:a5:11:2e:49:42 (capab=0x1511 status=0 aid=3)
[   13.071262] wlp0s20f3: associated
[   69.950023] Lockdown: systemd-logind: hibernation is restricted; see man kernel_lockdown.7
[   69.952359] Lockdown: systemd-logind: hibernation is restricted; see man kernel_lockdown.7
[   69.954024] Lockdown: systemd-logind: hibernation is restricted; see man kernel_lockdown.7
[   69.957794] Lockdown: systemd-logind: hibernation is restricted; see man kernel_lockdown.7
[   69.959130] Lockdown: systemd-logind: hibernation is restricted; see man kernel_lockdown.7
[   70.009219] Lockdown: systemd-logind: hibernation is restricted; see man kernel_lockdown.7
[   70.013624] Lockdown: systemd-logind: hibernation is restricted; see man kernel_lockdown.7
[   70.020654] Lockdown: systemd-logind: hibernation is restricted; see man kernel_lockdown.7
[   70.020936] wlp0s20f3: deauthenticating from bc:a5:11:2e:49:42 by local choice (Reason: 3=DEAUTH_LEAVING)
[   70.022786] Lockdown: systemd-logind: hibernation is restricted; see man kernel_lockdown.7
[   70.088816] Lockdown: systemd-logind: hibernation is restricted; see man kernel_lockdown.7
[   70.305973] warning: `iwconfig' uses wireless extensions which will stop working for Wi-Fi 7 hardware; use nl80211
[   71.345110] PM: suspend entry (s2idle)
[   71.349661] Filesystems sync: 0.004 seconds
[   71.366502] Freezing user space processes
[   71.368286] Freezing user space processes completed (elapsed 0.001 seconds)
[   71.368291] OOM killer disabled.
[   71.368292] Freezing remaining freezable tasks
[   71.369526] Freezing remaining freezable tasks completed (elapsed 0.001 seconds)
[   71.369529] printk: Suspending console(s) (use no_console_suspend to debug)
[   80.595829] surface_serial_hub serial0-0: pm: display-off notification failed: -110
[   80.595837] surface_serial_hub serial0-0: PM: device_prepare(): acpi_subsys_prepare+0x0/0x90 returns -110
[   80.595850] surface_serial_hub serial0-0: PM: not prepared for power transition: code -110
[   80.595854] PM: Some devices failed to suspend, or early wake event detected
[   80.596880] OOM killer enabled.
[   80.596883] Restarting tasks ... 
[   80.598453] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[   80.598463] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[   80.598551] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[   80.598625] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[   80.598791] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[   80.598797] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[   80.598880] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[   80.598966] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[   80.599055] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[   80.599523] surface_serial_hub serial0-0: rtl: dropping unexpected command message (rqid = 0x004f)
[   80.600993] surface_serial_hub serial0-0: rtl: dropping unexpected command message (rqid = 0x0050)
[   80.602299] done.
[   80.602320] random: crng reseeded on system resumption
[   80.602324] thermal thermal_zone0: failed to read out thermal zone (-61)
[   80.602338] surface_serial_hub serial0-0: rtl: dropping unexpected command message (rqid = 0x0051)
[   80.758480] PM: suspend exit
[   80.758626] PM: suspend entry (s2idle)
[   80.767079] Filesystems sync: 0.008 seconds
[   80.801065] Freezing user space processes
[   80.803776] Freezing user space processes completed (elapsed 0.002 seconds)
[   80.803785] OOM killer disabled.
[   80.803787] Freezing remaining freezable tasks
[   80.805136] Freezing remaining freezable tasks completed (elapsed 0.001 seconds)
[   80.805141] printk: Suspending console(s) (use no_console_suspend to debug)
[  101.834476] surface_serial_hub serial0-0: pm: D0-entry notification failed: -110
[  111.050482] surface_ac 01:02:01:01:01: PM: dpm_run_callback(): surface_ac_resume+0x0/0x60 [surface_charger] returns -110
[  111.050506] surface_ac 01:02:01:01:01: PM: failed to resume: error -110
[  120.266526] surface_battery 01:02:01:01:00: PM: dpm_run_callback(): surface_battery_resume+0x0/0x20 [surface_battery] returns -110
[  120.266549] surface_battery 01:02:01:01:00: PM: failed to resume: error -110
[  129.482537] surface_serial_hub serial0-0: pm: display-on notification failed: -110
[  129.483571] OOM killer enabled.
[  129.483574] Restarting tasks ... 
[  129.483623] mei_hdcp 0000:00:16.0-b638ab7e-94e2-4ea2-a552-d1c54b627f04: bound 0000:00:02.0 (ops i915_hdcp_ops [i915])
[  129.488837] done.
[  129.488858] random: crng reseeded on system resumption
[  129.488862] thermal thermal_zone0: failed to read out thermal zone (-61)
[  129.490637] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.490643] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.490644] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.490646] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.490647] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.490648] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.490650] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.490651] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.490658] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.490659] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.490661] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.490662] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.490663] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.490664] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.490665] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.490666] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.490668] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.490669] surface_serial_hub serial0-0: ptl: received ACK for non-pending packet
[  129.491983] surface_serial_hub serial0-0: rtl: dropping unexpected command message (rqid = 0x0054)
[  129.493848] surface_serial_hub serial0-0: rtl: dropping unexpected command message (rqid = 0x0055)
[  129.495179] surface_serial_hub serial0-0: rtl: dropping unexpected command message (rqid = 0x0056)
[  129.496213] surface_serial_hub serial0-0: rtl: dropping unexpected command message (rqid = 0x005d)
[  129.497271] surface_serial_hub serial0-0: rtl: dropping unexpected command message (rqid = 0x005e)
[  129.498317] surface_serial_hub serial0-0: rtl: dropping unexpected command message (rqid = 0x0057)
[  129.499390] surface_serial_hub serial0-0: rtl: dropping unexpected command message (rqid = 0x005c)
[  129.649199] PM: suspend exit
[  133.401214] wlp0s20f3: authenticate with c2:a5:11:2e:49:3f
[  133.401230] wlp0s20f3: 80 MHz not supported, disabling VHT
[  133.410119] wlp0s20f3: send auth to c2:a5:11:2e:49:3f (try 1/3)
[  133.474566] wlp0s20f3: authenticated
[  133.478495] wlp0s20f3: associate with c2:a5:11:2e:49:3f (try 1/3)
[  133.584974] wlp0s20f3: RX AssocResp from c2:a5:11:2e:49:3f (capab=0x1431 status=0 aid=3)
[  133.592652] wlp0s20f3: associated
[  153.863675] wlp0s20f3: disconnect from AP c2:a5:11:2e:49:3f for new auth to bc:a5:11:2e:49:42
[  153.916064] wlp0s20f3: authenticate with bc:a5:11:2e:49:42
[  153.920474] wlp0s20f3: send auth to bc:a5:11:2e:49:42 (try 1/3)
[  153.948926] wlp0s20f3: authenticated
[  153.950401] wlp0s20f3: associate with bc:a5:11:2e:49:42 (try 1/3)
[  154.050115] wlp0s20f3: RX ReassocResp from bc:a5:11:2e:49:42 (capab=0x1511 status=0 aid=3)
[  154.057267] wlp0s20f3: associated

```

</details>

**Comment by Omar-Abdelrady** [comment]
I have the same problem, Have u found any solution???

**Comment by Omar-Abdelrady** [comment]
https://github.com/linux-surface/linux-surface/issues/1258#issuecomment-1828882077

fixed bro @snizzo

**Comment by qzed** [comment]
If this is the same as #1258, then it should be fixed by (fairly) recent kernel updates. I assume they will trickle down to the generic Ubuntu kernels as well at some point.

**Comment by snizzo** [comment]
Nice! Will reopen if it's not fixed with 6.6 updates when it will come in generic ubuntu kernel. Thank you all!
